pkg/common/si_helper.go

```diff
@@ -19,16 +19,48 @@
 package common
 
 import (
+	"encoding/json"
+
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+
 	"github.com/apache/incubator-yunikorn-k8shim/pkg/conf"
+	"github.com/apache/incubator-yunikorn-k8shim/pkg/log"
 	"github.com/apache/incubator-yunikorn-scheduler-interface/lib/go/si"
 )
 
-func CreateUpdateRequestForTask(appID, taskID string, resource *si.Resource) si.UpdateRequest {
+func createTagsForTask(pod *v1.Pod, taskID string) map[string]string {
+	tags := map[string]string{
+		"namespace": pod.Namespace,
+		"podName": pod.Name,
+	}
+
+	// add Pod labels to Task tags
+	labels, err := json.Marshal(pod.Labels)
+	if err != nil {
+		log.Logger.Debug("Could not marshal pod labels", zap.String("taskID", taskID))
+	} else {
+		tags["labels"] = string(labels)
+	}
```

Comment:
One of earlier comment I have was, instead of marshaling this to a single string, can we keep the label values as key-value pairs. So in tags, we are not expecting
```
labels:  <json values for all labels>
```

instead, we'd like to have
```
kubernetes.io/labe/<some-label-key-1>: <some-value-2>
kubernetes.io/labe/<some-label-key-2>: <some-value-2>
```

this is to differentiate other tags from YK itself.

Reply:
Fixed.